### PR TITLE
DEV-115: WWW/HQ Entity Edit Filters

### PR DESCRIPTION
**Pull request by zfox23**
[DEV-115](https://highfidelity.atlassian.net/browse/DEV-115)

### CODE REVIEW

File: domainEntityFilters/www/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# `www` Entity Edit Filter 
+This is the Entity Edit Filter that goes in the `hifi://www` domain.
+
+## Parameters
+This Entity Edit Filter has the following parameters:
+1. All PHYSICS edits are disallowed.
+2. When the ES receives an entity ADD packet, the filter will only allow entities to be added if the name of the new entity contains "Whiteboard".
+    - This allows the on-board Whiteboard Polylines to get added to the ES' entity tree.
+    - Theoretically, this means anyone could add an entity to the tree if they named the entity with this knowledge, but we don't expect people to grief their headquarters on purpose :).
+3. When the ES receives an entity EDIT packet, the filter will only allow entities to be added if the original name of the entity contains "Whiteboard". Otherwise, the filter will allow all edits to `privateUserData`; the entity server will handle authenticating users to verify that they're actually allowed to edit `privateUserData`.
```
Comment (MiladNazeri) [inline domainEntityFilters/www/README.md]: Should that be allow entities to be edited in #3?

File: domainEntityFilters/www/filter.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+//  Created by Zach Fox on 2019-06-07
+//  Copyright 2019 High Fidelity, Inc.
+//
+//  See accompanying README.md for usage instructions.
+// 
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+
+
+function filterAdd(properties, originalProperties) {
```
Comment (MiladNazeri) [inline domainEntityFilters/www/filter.js]: Fitler add doesn't have original properties btw

File: domainEntityFilters/www/filter.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    return false;
+}
+
+
+function filterEdit(properties, originalProperties) {
+    if (originalProperties.name.indexOf("Whiteboard") > -1) {
+        return properties;
+    } else if (properties.privateUserData) {
+        var retProps = {
+            privateUserData: properties.privateUserData
```
Comment (MiladNazeri) [inline domainEntityFilters/www/filter.js]: Have you had a chance to test if this works as expected?

File: domainEntityFilters/www/filter.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+
+var FILTER_TYPE_ADD = 0;
+var FILTER_TYPE_EDIT = 1;
+var FILTER_TYPE_PHYSICS = 2;
+var FILTER_TYPE_DELETE = 3;
+function filter(properties, filterType, originalProperties) {
+    switch (filterType) {
+        case FILTER_TYPE_ADD:
+            return filterAdd(properties, originalProperties);
```
Comment (MiladNazeri) [inline domainEntityFilters/www/filter.js]: note above about original properties

File: domainEntityFilters/www/filter.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        case FILTER_TYPE_EDIT:
+            return filterEdit(properties, originalProperties);
+        case FILTER_TYPE_PHYSICS:
+            return false;
+        case FILTER_TYPE_DELETE:
+            return filterDelete(properties, originalProperties);
```
Comment (MiladNazeri) [inline domainEntityFilters/www/filter.js]: note above about original properties

File: domainEntityFilters/www/filter.js (hunk 3 above)
Comment (zfox23) [reply domainEntityFilters/www/filter.js]: yup! and you helped! :D ty!

File: domainEntityFilters/www/filter.js (hunk 2 above)
Comment (zfox23) [reply domainEntityFilters/www/filter.js]: ack!

File: domainEntityFilters/www/README.md (hunk 1 above)
Comment (zfox23) [reply domainEntityFilters/www/README.md]: yes, thank you!

File: domainEntityFilters/www/filter.js (hunk 4 above)
Comment (zfox23) [reply domainEntityFilters/www/filter.js]: ack!

File: domainEntityFilters/www/filter.js (hunk 5 above)
Comment (zfox23) [reply domainEntityFilters/www/filter.js]: i disagree - `originalProperties` are necessary when deleting! (i tested this to verify)

**Pull request merged**